homeassistant/components/advantage_air/climate.py

```diff
@@ -188,6 +210,54 @@ async def async_set_temperature(self, **kwargs: Any) -> None:
                 }
             )
 
+    async def async_set_preset_mode(self, preset_mode: str) -> None:
+        """Set the preset mode."""
+        change = {}
+        if ADVANTAGE_AIR_MYTEMP_ENABLED in self._ac:
+            change[ADVANTAGE_AIR_MYTEMP_ENABLED] = preset_mode == ADVANTAGE_AIR_MYTEMP
+        if ADVANTAGE_AIR_MYAUTO_ENABLED in self._ac:
+            change[ADVANTAGE_AIR_MYAUTO_ENABLED] = preset_mode == ADVANTAGE_AIR_MYAUTO
+        await self.async_update_ac(change)
+
+    async def async_added_to_hass(self) -> None:
+        """Subscribe to state changes if required."""
+        await super().async_added_to_hass()
+        # If presets are supported
+        if self._attr_supported_features & ClimateEntityFeature.PRESET_MODE:
+            # Track state changes, as preset changes require a reload
+            self.async_on_remove(
+                async_track_state_change_event(
+                    self.hass, self.entity_id, self.check_preset
+                )
+            )
+
+    @callback
+    async def check_preset(self, event: Event) -> None:
```

Comment:
A coroutine function can't be a callback. If we don't need to await inside this can just be a callback.